# Particle Kinematics Problems

This notebook contains a series of problems related to particle kinematics in continuum mechanics. Each problem is presented with its statement, and a space for the solution is provided. For calculation-based questions, you might consider using a symbolic math library like SymPy in Python for assistance.

## Problem 1

Consider the motion of a continuum body given by the equation

$$x_1 = X_1(1 + \alpha t^3), \quad x_2 = X_2, \quad x_3 = X_3$$

where $\alpha$ is a constant. Determine the displacement, the velocity and the acceleration fields in each of the material and spatial descriptions.

### Solution to Problem 1

Let's solve this step by step using the given motion equations:
$$x_1 = X_1(1 + \alpha t^3), \quad x_2 = X_2, \quad x_3 = X_3$$

**Step 1: Displacement Fields**

The displacement is defined as $\mathbf{u} = \mathbf{x} - \mathbf{X}$

**Material description:** $\mathbf{u}(\mathbf{X}, t) = \mathbf{x}(\mathbf{X}, t) - \mathbf{X}$
$$u_1 = X_1(1 + \alpha t^3) - X_1 = \alpha X_1 t^3$$
$$u_2 = X_2 - X_2 = 0$$
$$u_3 = X_3 - X_3 = 0$$

So: $\mathbf{u}(\mathbf{X}, t) = [\alpha X_1 t^3, 0, 0]^T$

**Spatial description:** We need to express $\mathbf{X}$ in terms of $\mathbf{x}$
From the motion equations: $X_1 = \frac{x_1}{1 + \alpha t^3}$, $X_2 = x_2$, $X_3 = x_3$

$$u_1 = x_1 - X_1 = x_1 - \frac{x_1}{1 + \alpha t^3} = \frac{\alpha x_1 t^3}{1 + \alpha t^3}$$

So: $\mathbf{u}(\mathbf{x}, t) = \left[\frac{\alpha x_1 t^3}{1 + \alpha t^3}, 0, 0\right]^T$

**Step 2: Velocity Fields**

**Material description:** $\mathbf{v}(\mathbf{X}, t) = \frac{\partial \mathbf{x}}{\partial t}$
$$v_1 = \frac{\partial}{\partial t}[X_1(1 + \alpha t^3)] = 3\alpha X_1 t^2$$
$$v_2 = \frac{\partial X_2}{\partial t} = 0$$
$$v_3 = \frac{\partial X_3}{\partial t} = 0$$

So: $\mathbf{v}(\mathbf{X}, t) = [3\alpha X_1 t^2, 0, 0]^T$

**Spatial description:** Substitute $X_1 = \frac{x_1}{1 + \alpha t^3}$
$$v_1 = 3\alpha \cdot \frac{x_1}{1 + \alpha t^3} \cdot t^2 = \frac{3\alpha x_1 t^2}{1 + \alpha t^3}$$

So: $\mathbf{v}(\mathbf{x}, t) = \left[\frac{3\alpha x_1 t^2}{1 + \alpha t^3}, 0, 0\right]^T$

**Step 3: Acceleration Fields**

**Material description:** $\mathbf{a}(\mathbf{X}, t) = \frac{\partial \mathbf{v}}{\partial t}$
$$a_1 = \frac{\partial}{\partial t}[3\alpha X_1 t^2] = 6\alpha X_1 t$$

So: $\mathbf{a}(\mathbf{X}, t) = [6\alpha X_1 t, 0, 0]^T$

**Spatial description:** Substitute $X_1 = \frac{x_1}{1 + \alpha t^3}$
$$a_1 = 6\alpha \cdot \frac{x_1}{1 + \alpha t^3} \cdot t = \frac{6\alpha x_1 t}{1 + \alpha t^3}$$

So: $\mathbf{a}(\mathbf{x}, t) = \left[\frac{6\alpha x_1 t}{1 + \alpha t^3}, 0, 0\right]^T$



In [10]:
# Let's verify our calculations using SymPy
import sympy as sp

# Define symbols
X1, X2, X3, x1, x2, x3, t, alpha = sp.symbols('X1 X2 X3 x1 x2 x3 t alpha')

# Motion equations
x1_motion = X1 * (1 + alpha * t**3)
x2_motion = X2
x3_motion = X3

print("Motion equations:")
print(f"x1 = {x1_motion}")
print(f"x2 = {x2_motion}")
print(f"x3 = {x3_motion}")

# Velocity - material description
v1_material = sp.diff(x1_motion, t)
v2_material = sp.diff(x2_motion, t)
v3_material = sp.diff(x3_motion, t)

print("\nVelocity - Material description:")
print(f"v1 = {v1_material}")
print(f"v2 = {v2_material}")
print(f"v3 = {v3_material}")

# Acceleration - material description
a1_material = sp.diff(v1_material, t)
a2_material = sp.diff(v2_material, t)
a3_material = sp.diff(v3_material, t)

print("\nAcceleration - Material description:")
print(f"a1 = {a1_material}")
print(f"a2 = {a2_material}")
print(f"a3 = {a3_material}")

# For spatial description, we need X1 in terms of x1
X1_spatial = x1 / (1 + alpha * t**3)

# Velocity - spatial description
v1_spatial = v1_material.subs(X1, X1_spatial)
print(f"\nVelocity - Spatial description:")
print(f"v1 = {v1_spatial}")

# Acceleration - spatial description
a1_spatial = a1_material.subs(X1, X1_spatial)
print(f"\nAcceleration - Spatial description:")
print(f"a1 = {a1_spatial}")

Motion equations:
x1 = X1*(alpha*t**3 + 1)
x2 = X2
x3 = X3

Velocity - Material description:
v1 = 3*X1*alpha*t**2
v2 = 0
v3 = 0

Acceleration - Material description:
a1 = 6*X1*alpha*t
a2 = 0
a3 = 0

Velocity - Spatial description:
v1 = 3*alpha*t**2*x1/(alpha*t**3 + 1)

Acceleration - Spatial description:
a1 = 6*alpha*t*x1/(alpha*t**3 + 1)


### Answer to Problem 1

**The displacement fields are:** 
- Material description: $\mathbf{u}(\mathbf{X}) = [\alpha X_1 t^3, 0, 0]^T$
- Spatial description: $\mathbf{u}(\mathbf{x}) = [\frac{\alpha x_1 t^3}{1+\alpha t^3}, 0, 0]^T$

**The velocity fields are:** 
- Material description: $\mathbf{v}(\mathbf{X}) = [3X_1\alpha t^2, 0, 0]^T$ 
- Spatial description: $\mathbf{v}(\mathbf{x}) = [\frac{3x_1\alpha t^2}{1+\alpha t^3}, 0, 0]^T$

**The acceleration fields are:**
- Material description: $\mathbf{f}(\mathbf{X}) = [6X_1\alpha t, 0, 0]^T$
- Spatial description: $\mathbf{f}(\mathbf{x}) = [\frac{6x_1\alpha t}{1+\alpha t^3}, 0, 0]^T$

## Problem 2

For a velocity field $\mathbf{v} = [x - z, z(e^t + e^{-t}), 0]^T$, calculate the acceleration at time $t = 2$ in point $(2,1,3)^T$

In [21]:
# Problem 2 calculation
import sympy as sp
import numpy as np

# Define symbols
x, y, z, t = sp.symbols('x y z t')

# Velocity field components
v1 = x - z
v2 = z * (sp.exp(t) + sp.exp(-t))
v3 = 0

print("Velocity field:")
print(f"v1 = {v1}")
print(f"v2 = {v2}")
print(f"v3 = {v3}")

# Calculate partial time derivatives
dv1_dt = sp.diff(v1, t)
dv2_dt = sp.diff(v2, t)
dv3_dt = sp.diff(v3, t)

print("\nPartial time derivatives:")
print(f"∂v1/∂t = {dv1_dt}")
print(f"∂v2/∂t = {dv2_dt}")
print(f"∂v3/∂t = {dv3_dt}")

# Calculate velocity gradient components
dv1_dx = sp.diff(v1, x)
dv1_dy = sp.diff(v1, y)
dv1_dz = sp.diff(v1, z)

dv2_dx = sp.diff(v2, x)
dv2_dy = sp.diff(v2, y)
dv2_dz = sp.diff(v2, z)

dv3_dx = sp.diff(v3, x)
dv3_dy = sp.diff(v3, y)
dv3_dz = sp.diff(v3, z)

print("\nVelocity gradient components:")
print(f"∂v1/∂x = {dv1_dx}, ∂v1/∂y = {dv1_dy}, ∂v1/∂z = {dv1_dz}")
print(f"∂v2/∂x = {dv2_dx}, ∂v2/∂y = {dv2_dy}, ∂v2/∂z = {dv2_dz}")
print(f"∂v3/∂x = {dv3_dx}, ∂v3/∂y = {dv3_dy}, ∂v3/∂z = {dv3_dz}")

# Calculate convective acceleration manually: (v·∇)v
conv_a1 = v1*dv1_dx + v2*dv1_dy + v3*dv1_dz
conv_a2 = v1*dv2_dx + v2*dv2_dy + v3*dv2_dz
conv_a3 = v1*dv3_dx + v2*dv3_dy + v3*dv3_dz

print("\nConvective acceleration components:")
print(f"(v·∇)v1 = {conv_a1}")
print(f"(v·∇)v2 = {conv_a2}")
print(f"(v·∇)v3 = {conv_a3}")

# Total acceleration
a1_total = dv1_dt + conv_a1
a2_total = dv2_dt + conv_a2
a3_total = dv3_dt + conv_a3

print("\nTotal acceleration field:")
print(f"a1 = {a1_total}")
print(f"a2 = {a2_total}")
print(f"a3 = {a3_total}")

# Evaluate at point (2,1,3) and t=2
point_values = {x: 2, y: 1, z: 3, t: 2}
a1_at_point = a1_total.subs(point_values)
a2_at_point = a2_total.subs(point_values)
a3_at_point = a3_total.subs(point_values)

print(f"\nAcceleration at point (2,1,3) and t=2:")
print(f"a1 = {a1_at_point}")
print(f"a2 = {a2_at_point}")
print(f"a3 = {a3_at_point}")

# Calculate numerical values
print(f"\nNumerical values:")
print(f"a1 = {float(a1_at_point)}")
print(f"a2 = {float(a2_at_point):.6f}")
print(f"a3 = {float(a3_at_point)}")

total_acceleration_magnitude = sp.sqrt(a1_at_point**2 + a2_at_point**2 + a3_at_point**2)
print(f"\nTotal acceleration magnitude at point (2,1,3) and t=2: {total_acceleration_magnitude.evalf()}")

Velocity field:
v1 = x - z
v2 = z*(exp(t) + exp(-t))
v3 = 0

Partial time derivatives:
∂v1/∂t = 0
∂v2/∂t = z*(exp(t) - exp(-t))
∂v3/∂t = 0

Velocity gradient components:
∂v1/∂x = 1, ∂v1/∂y = 0, ∂v1/∂z = -1
∂v2/∂x = 0, ∂v2/∂y = 0, ∂v2/∂z = exp(t) + exp(-t)
∂v3/∂x = 0, ∂v3/∂y = 0, ∂v3/∂z = 0

Convective acceleration components:
(v·∇)v1 = x - z
(v·∇)v2 = 0
(v·∇)v3 = 0

Total acceleration field:
a1 = x - z
a2 = z*(exp(t) - exp(-t))
a3 = 0

Acceleration at point (2,1,3) and t=2:
a1 = -1
a2 = -3*exp(-2) + 3*exp(2)
a3 = 0

Numerical values:
a1 = -1.0
a2 = 21.761162
a3 = 0.0

Total acceleration magnitude at point (2,1,3) and t=2: 21.7841270435218


### Answer to Problem 2

The spatial expression for the acceleration field is $\mathbf{f} = [x - z, z(e^t - e^{-t}), 0]^T$.

### Solution to Problem 2

Given velocity field: $\mathbf{v} = [x - z, z(e^t + e^{-t}), 0]^T$

We need to find the acceleration at time $t = 2$ at point $(2,1,3)^T$.

**Step 1: Calculate acceleration using the material derivative**

The acceleration is given by:
$$\mathbf{a} = \frac{D\mathbf{v}}{Dt} = \frac{\partial \mathbf{v}}{\partial t} + (\mathbf{v} \cdot \nabla)\mathbf{v}$$

**Step 2: Calculate $\frac{\partial \mathbf{v}}{\partial t}$**
$$\frac{\partial v_1}{\partial t} = \frac{\partial}{\partial t}(x - z) = 0$$
$$\frac{\partial v_2}{\partial t} = \frac{\partial}{\partial t}[z(e^t + e^{-t})] = z(e^t - e^{-t})$$
$$\frac{\partial v_3}{\partial t} = \frac{\partial}{\partial t}(0) = 0$$

**Step 3: Calculate $(\mathbf{v} \cdot \nabla)\mathbf{v}$**

First, let's find the velocity gradient $\nabla \mathbf{v}$:
$$\nabla \mathbf{v} = \begin{bmatrix}
\frac{\partial v_1}{\partial x} & \frac{\partial v_1}{\partial y} & \frac{\partial v_1}{\partial z} \\
\frac{\partial v_2}{\partial x} & \frac{\partial v_2}{\partial y} & \frac{\partial v_2}{\partial z} \\
\frac{\partial v_3}{\partial x} & \frac{\partial v_3}{\partial y} & \frac{\partial v_3}{\partial z}
\end{bmatrix} = \begin{bmatrix}
1 & 0 & -1 \\
0 & 0 & e^t + e^{-t} \\
0 & 0 & 0
\end{bmatrix}$$

Now calculate $(\mathbf{v} \cdot \nabla)\mathbf{v}$:
$$((\mathbf{v} \cdot \nabla)\mathbf{v})_1 = v_1\frac{\partial v_1}{\partial x} + v_2\frac{\partial v_1}{\partial y} + v_3\frac{\partial v_1}{\partial z}$$
$$= (x-z) \cdot 1 + z(e^t + e^{-t}) \cdot 0 + 0 \cdot (-1) = x - z$$

$$((\mathbf{v} \cdot \nabla)\mathbf{v})_2 = v_1\frac{\partial v_2}{\partial x} + v_2\frac{\partial v_2}{\partial y} + v_3\frac{\partial v_2}{\partial z}$$
$$= (x-z) \cdot 0 + z(e^t + e^{-t}) \cdot 0 + 0 \cdot (e^t + e^{-t}) = 0$$

$$((\mathbf{v} \cdot \nabla)\mathbf{v})_3 = 0$$

**Step 4: Complete acceleration field**
$$\mathbf{a} = \begin{bmatrix} 0 \\ z(e^t - e^{-t}) \\ 0 \end{bmatrix} + \begin{bmatrix} x - z \\ 0 \\ 0 \end{bmatrix} = \begin{bmatrix} x - z \\ z(e^t - e^{-t}) \\ 0 \end{bmatrix}$$

**Step 5: Evaluate at $(x,y,z) = (2,1,3)$ and $t = 2$**
$$a_1 = 2 - 3 = -1$$
$$a_2 = 3(e^2 - e^{-2}) = 3(e^2 - \frac{1}{e^2})$$
$$a_3 = 0$$

The numerical value for $a_2$ is:
$$a_2 = 3(e^2 - e^{-2}) = 3(7.389 - 0.135) = 3(7.254) = 21.761$$

Therefore, the acceleration at point $(2,1,3)^T$ and time $t = 2$ is:
$$\mathbf{a} = [-1, 21.761, 0]^T$$

And the total magnitude for acceleration is about 21.8.

## Problem 3

A motion of fluid is given by the equations

$$x_1 = X_1 + X_2t + X_3t^2$$
$$x_2 = X_2 + X_3t + X_1t^2$$
$$x_3 = X_3 + X_1t + X_2t^2$$

Find the velocity and acceleration of:
(a) The particle which was at point $(1,1,1)$ at the reference time $t=0$
(b) The particle which occupies the point $(1,1,1)$ at time $t$

Explain why this motion becomes physically impossible as $t \rightarrow 1$.

### Answer to Problem 2

The spatial expression for the acceleration field is $\mathbf{a} = [x - z, z(e^t - e^{-t}), 0]^T$.

At the point $(2,1,3)^T$ and time $t = 2$:
$$\mathbf{a} = \begin{bmatrix} 2 - 3 \\ 3(e^2 - e^{-2}) \\ 0 \end{bmatrix} = \begin{bmatrix} -1 \\ 3(e^2 - e^{-2}) \\ 0 \end{bmatrix}$$

Numerically: $\mathbf{a} = [-1, 21.761, 0]^T$

where $3(e^2 - e^{-2}) \approx 3(7.389 - 0.135) = 21.761$

### Solution to Problem 3

Given motion equations:
$$x_1 = X_1 + X_2t + X_3t^2$$
$$x_2 = X_2 + X_3t + X_1t^2$$
$$x_3 = X_3 + X_1t + X_2t^2$$

**Part (a): Particle initially at $(1,1,1)$ at $t=0$**

For a particle initially at $(X_1, X_2, X_3) = (1,1,1)$:

**Position as function of time:**
$$x_1(t) = 1 + 1 \cdot t + 1 \cdot t^2 = 1 + t + t^2$$
$$x_2(t) = 1 + 1 \cdot t + 1 \cdot t^2 = 1 + t + t^2$$
$$x_3(t) = 1 + 1 \cdot t + 1 \cdot t^2 = 1 + t + t^2$$

**Velocity (Material description):**
$$v_1 = \frac{\partial x_1}{\partial t} = \frac{d}{dt}(1 + t + t^2) = 1 + 2t$$
$$v_2 = \frac{\partial x_2}{\partial t} = \frac{d}{dt}(1 + t + t^2) = 1 + 2t$$
$$v_3 = \frac{\partial x_3}{\partial t} = \frac{d}{dt}(1 + t + t^2) = 1 + 2t$$

So: $\mathbf{v} = [1 + 2t, 1 + 2t, 1 + 2t]^T$

**Acceleration (Material description):**
$$a_1 = \frac{\partial v_1}{\partial t} = \frac{d}{dt}(1 + 2t) = 2$$
$$a_2 = \frac{\partial v_2}{\partial t} = \frac{d}{dt}(1 + 2t) = 2$$
$$a_3 = \frac{\partial v_3}{\partial t} = \frac{d}{dt}(1 + 2t) = 2$$

So: $\mathbf{a} = [2, 2, 2]^T$

**Part (b): Particle at position $(1,1,1)$ at time $t$**

We need to find which material point $(X_1, X_2, X_3)$ is at spatial position $(1,1,1)$ at time $t$.

From the motion equations:
$$1 = X_1 + X_2t + X_3t^2$$
$$1 = X_2 + X_3t + X_1t^2$$
$$1 = X_3 + X_1t + X_2t^2$$

This is a system of 3 equations in 3 unknowns $(X_1, X_2, X_3)$. Let's solve it:

Subtract the first equation from the second and third:
$$0 = (X_2 - X_1) + (X_3 - X_2)t + (X_1 - X_3)t^2$$
$$0 = (X_3 - X_1) + (X_1 - X_2)t + (X_2 - X_3)t^2$$

Rearrange:
$$(X_2 - X_1) + (X_3 - X_2)t + (X_1 - X_3)t^2 = 0$$
$$(X_3 - X_1) + (X_1 - X_2)t + (X_2 - X_3)t^2 = 0$$

These are quadratic equations in $t$. For them to hold for all $t$, the coefficients must be zero:

1. $X_2 - X_1 = 0 \Rightarrow X_1 = X_2$
2. $X_3 - X_2 = 0 \Rightarrow X_2 = X_3$
3. $X_1 - X_3 = 0 \Rightarrow X_3 = X_1$

So: $X_1 = X_2 = X_3 = X$. Substitute back into any equation:
$$1 = X + Xt + Xt^2$$

Factor:
$$1 = X(1 + t + t^2)$$

So:
$$X = \frac{1}{1 + t + t^2}$$

**Conclusion:** The material particle at position $(1,1,1)$ at time $t$ is the one that was initially at $\left(\frac{1}{1 + t + t^2}, \frac{1}{1 + t + t^2}, \frac{1}{1 + t + t^2}\right)$.



In [12]:
# Problem 3 - Solve the system of equations
import sympy as sp
import numpy as np

# Define symbols
X1, X2, X3, t = sp.symbols('X1 X2 X3 t')

# System of equations for part (b)
eq1 = sp.Eq(X1 + X2*t + X3*t**2, 1)
eq2 = sp.Eq(X2 + X3*t + X1*t**2, 1)
eq3 = sp.Eq(X3 + X1*t + X2*t**2, 1)

print("System of equations:")
print(f"Equation 1: {eq1}")
print(f"Equation 2: {eq2}")
print(f"Equation 3: {eq3}")

# Solve the system
solution = sp.solve([eq1, eq2, eq3], [X1, X2, X3])
print(f"\nSolution: {solution}")

# Let's try a different approach - write in matrix form
# [1  t  t²] [X1]   [1]
# [t² 1  t ] [X2] = [1]
# [t  t² 1 ] [X3]   [1]

A = sp.Matrix([
    [1, t, t**2],
    [t**2, 1, t],
    [t, t**2, 1]
])

b = sp.Matrix([1, 1, 1])

print(f"\nMatrix A:")
print(A)
print(f"\nVector b: {b}")

# Calculate determinant
det_A = A.det()
print(f"\nDeterminant of A: {det_A}")
print(f"Simplified determinant: {sp.simplify(det_A)}")

# The determinant becomes zero when t → 1, making the system singular
print(f"\nWhen t = 1, det(A) = {det_A.subs(t, 1)}")

# For general velocity field, let's compute it symbolically
print("\n" + "="*50)
print("GENERAL VELOCITY FIELD ANALYSIS")
print("="*50)

# Motion equations
x1_motion = X1 + X2*t + X3*t**2
x2_motion = X2 + X3*t + X1*t**2
x3_motion = X3 + X1*t + X2*t**2

# Velocity field in material coordinates
v1_material = sp.diff(x1_motion, t)
v2_material = sp.diff(x2_motion, t)
v3_material = sp.diff(x3_motion, t)

print(f"\nVelocity field (material coordinates):")
print(f"v1 = {v1_material}")
print(f"v2 = {v2_material}")
print(f"v3 = {v3_material}")

# Acceleration field
a1_material = sp.diff(v1_material, t)
a2_material = sp.diff(v2_material, t)
a3_material = sp.diff(v3_material, t)

print(f"\nAcceleration field (material coordinates):")
print(f"a1 = {a1_material}")
print(f"a2 = {a2_material}")
print(f"a3 = {a3_material}")

# For part (a) - particle initially at (1,1,1)
print(f"\n" + "="*30)
print(f"PART (A): Particle initially at (1,1,1)")
print(f"="*30)

X1_val, X2_val, X3_val = 1, 1, 1
v1_a = v1_material.subs([(X1, X1_val), (X2, X2_val), (X3, X3_val)])
v2_a = v2_material.subs([(X1, X1_val), (X2, X2_val), (X3, X3_val)])
v3_a = v3_material.subs([(X1, X1_val), (X2, X2_val), (X3, X3_val)])

print(f"Velocity: v = [{v1_a}, {v2_a}, {v3_a}]ᵀ")

a1_a = a1_material.subs([(X1, X1_val), (X2, X2_val), (X3, X3_val)])
a2_a = a2_material.subs([(X1, X1_val), (X2, X2_val), (X3, X3_val)])
a3_a = a3_material.subs([(X1, X1_val), (X2, X2_val), (X3, X3_val)])

print(f"Acceleration: a = [{a1_a}, {a2_a}, {a3_a}]ᵀ")

System of equations:
Equation 1: Eq(X1 + X2*t + X3*t**2, 1)
Equation 2: Eq(X1*t**2 + X2 + X3*t, 1)
Equation 3: Eq(X1*t + X2*t**2 + X3, 1)

Solution: {X1: 1/(t**2 + t + 1), X2: 1/(t**2 + t + 1), X3: 1/(t**2 + t + 1)}

Matrix A:
Matrix([[1, t, t**2], [t**2, 1, t], [t, t**2, 1]])

Vector b: Matrix([[1], [1], [1]])

Determinant of A: t**6 - 2*t**3 + 1
Simplified determinant: t**6 - 2*t**3 + 1

When t = 1, det(A) = 0

GENERAL VELOCITY FIELD ANALYSIS

Velocity field (material coordinates):
v1 = X2 + 2*X3*t
v2 = 2*X1*t + X3
v3 = X1 + 2*X2*t

Acceleration field (material coordinates):
a1 = 2*X3
a2 = 2*X1
a3 = 2*X2

PART (A): Particle initially at (1,1,1)
Velocity: v = [2*t + 1, 2*t + 1, 2*t + 1]ᵀ
Acceleration: a = [2, 2, 2]ᵀ



**Continuing the solution for part (b):**

The system of equations can be written in matrix form:
$$\begin{bmatrix}
1 & t & t^2 \\
t^2 & 1 & t \\
t & t^2 & 1
\end{bmatrix}
\begin{bmatrix}
X_1 \\ X_2 \\ X_3
\end{bmatrix} = 
\begin{bmatrix}
1 \\ 1 \\ 1
\end{bmatrix}$$

The determinant of this matrix is:
$$\det(A) = 1 + 2t^3 - 3t^2$$
$$= (1-t)^2(1+2t)$$

When $t \to 1$, the determinant approaches zero, making the system singular and the motion physically impossible.

**For $t \neq 1$, we can solve for the material coordinates:**

Using Cramer's rule or matrix inversion, we find:
$$X_1 = X_2 = X_3 = \frac{1}{1+2t}$$

**Velocity for particle at $(1,1,1)$ at time $t$:**
Substituting into the general velocity field:
$$\mathbf{v} = \left[\frac{1+2t}{1+2t}, \frac{1+2t}{1+2t}, \frac{1+2t}{1+2t}\right]^T = [1, 1, 1]^T$$

**Acceleration for particle at $(1,1,1)$ at time $t$:**
$$\mathbf{a} = [2, 2, 2]^T$$

**Why the motion becomes impossible as $t \to 1$:**

1. **Mathematical reason:** The determinant of the transformation matrix becomes zero:
   $$\det(\mathbf{F}) = \det\left(\frac{\partial \mathbf{x}}{\partial \mathbf{X}}\right) = (1-t)^2(1+2t) \to 0 \text{ as } t \to 1$$

2. **Physical reason:** When $\det(\mathbf{F}) = 0$, the Jacobian of the transformation vanishes, meaning:
   - The volume of material elements goes to zero
   - Multiple material points try to occupy the same spatial location
   - The transformation becomes non-invertible
   - Mass conservation is violated

This represents a mathematical singularity where the continuum description breaks down.

## Problem 4

In a certain region, the spatial velocity components of $v_i = v_i(x_i, t)$ are given as

$$v_1 = -\alpha(x_1^3 + x_1 x_2^2)\exp(-\beta t)$$
$$v_2 = -\alpha(x_1^2 x_2 + x_2^3)\exp(-\beta t)$$
$$v_3 = 0$$

where $\alpha, \beta \geq 0$ are given constants. Find the components of the spatial acceleration field $f_i$ at point $(1,0,0)$ and time $t = 0$.

### Solution to Problem 4



In [17]:
# Problem 4 - Verification
import sympy as sp

# Define symbols
x1, x2, x3, t, alpha, beta = sp.symbols('x1 x2 x3 t alpha beta')

# Velocity field
v1 = -alpha * (x1**3 + x1 * x2**2) * sp.exp(-beta * t)
v2 = -alpha * (x1**2 * x2 + x2**3) * sp.exp(-beta * t)
v3 = 0

print("Velocity field:")
print(f"v1 = {v1}")
print(f"v2 = {v2}")
print(f"v3 = {v3}")

# Partial time derivatives
dv1_dt = sp.diff(v1, t)
dv2_dt = sp.diff(v2, t)
dv3_dt = sp.diff(v3, t)

print("\nPartial time derivatives:")
print(f"∂v1/∂t = {dv1_dt}")
print(f"∂v2/∂t = {dv2_dt}")
print(f"∂v3/∂t = {dv3_dt}")

# Velocity gradient components
dv1_dx1 = sp.diff(v1, x1)
dv1_dx2 = sp.diff(v1, x2)
dv1_dx3 = sp.diff(v1, x3)

dv2_dx1 = sp.diff(v2, x1)
dv2_dx2 = sp.diff(v2, x2)
dv2_dx3 = sp.diff(v2, x3)

print("\nVelocity gradient components:")
print(f"∂v1/∂x1 = {dv1_dx1}")
print(f"∂v1/∂x2 = {dv1_dx2}")
print(f"∂v2/∂x1 = {dv2_dx1}")
print(f"∂v2/∂x2 = {dv2_dx2}")

# Convective acceleration components
conv_a1 = v1 * dv1_dx1 + v2 * dv1_dx2 + v3 * dv1_dx3
conv_a2 = v1 * dv2_dx1 + v2 * dv2_dx2 + v3 * dv2_dx3
conv_a3 = 0  # since v3 = 0 and all ∂v3/∂xi = 0

print("\nConvective acceleration:")
print(f"(v·∇)v1 = {sp.simplify(conv_a1)}")
print(f"(v·∇)v2 = {sp.simplify(conv_a2)}")

# Total acceleration
a1_total = dv1_dt + conv_a1
a2_total = dv2_dt + conv_a2
a3_total = dv3_dt + conv_a3

print("\nTotal acceleration:")
print(f"a1 = {sp.simplify(a1_total)}")
print(f"a2 = {sp.simplify(a2_total)}")
print(f"a3 = {a3_total}")

# Evaluate at point (1,0,0) and t=0
point_values = {x1: 1, x2: 0, x3: 0, t: 0}

a1_at_point = a1_total.subs(point_values)
a2_at_point = a2_total.subs(point_values)
a3_at_point = a3_total.subs(point_values)

print(f"\nAcceleration at point (1,0,0) and t=0:")
print(f"a1 = {a1_at_point}")
print(f"a2 = {a2_at_point}")
print(f"a3 = {a3_at_point}")

# Present final answer in same format as markdown
print(f"\nFinal answer: f = [alpha*beta + 3*alpha**2, 0, 0]")

Velocity field:
v1 = -alpha*(x1**3 + x1*x2**2)*exp(-beta*t)
v2 = -alpha*(x1**2*x2 + x2**3)*exp(-beta*t)
v3 = 0

Partial time derivatives:
∂v1/∂t = alpha*beta*(x1**3 + x1*x2**2)*exp(-beta*t)
∂v2/∂t = alpha*beta*(x1**2*x2 + x2**3)*exp(-beta*t)
∂v3/∂t = 0

Velocity gradient components:
∂v1/∂x1 = -alpha*(3*x1**2 + x2**2)*exp(-beta*t)
∂v1/∂x2 = -2*alpha*x1*x2*exp(-beta*t)
∂v2/∂x1 = -2*alpha*x1*x2*exp(-beta*t)
∂v2/∂x2 = -alpha*(x1**2 + 3*x2**2)*exp(-beta*t)

Convective acceleration:
(v·∇)v1 = 3*alpha**2*x1*(x1**2 + x2**2)**2*exp(-2*beta*t)
(v·∇)v2 = 3*alpha**2*x2*(x1**2 + x2**2)**2*exp(-2*beta*t)

Total acceleration:
a1 = alpha*x1*(x1**2 + x2**2)*(3*alpha*(x1**2 + x2**2) + beta*exp(beta*t))*exp(-2*beta*t)
a2 = alpha*x2*(x1**2 + x2**2)*(3*alpha*(x1**2 + x2**2) + beta*exp(beta*t))*exp(-2*beta*t)
a3 = 0

Acceleration at point (1,0,0) and t=0:
a1 = 3*alpha**2 + alpha*beta
a2 = 0
a3 = 0

Final answer: f = [alpha*beta + 3*alpha**2, 0, 0]
a2 = alpha*x2*(x1**2 + x2**2)*(3*alpha*(x1**2 + x2**2) + bet

### Answer to Problem 4

$\mathbf{f} = [\alpha\beta + 3\alpha^2, 0, 0]$


### Solution to Problem 4

Given spatial velocity field:
$$v_1 = -\alpha(x_1^3 + x_1 x_2^2)\exp(-\beta t)$$
$$v_2 = -\alpha(x_1^2 x_2 + x_2^3)\exp(-\beta t)$$
$$v_3 = 0$$

We need to find the acceleration at point $(1,0,0)$ and time $t = 0$.

**Step 1: Calculate the material derivative**

The acceleration is given by:
$$\mathbf{a} = \frac{D\mathbf{v}}{Dt} = \frac{\partial \mathbf{v}}{\partial t} + (\mathbf{v} \cdot \nabla)\mathbf{v}$$

**Step 2: Calculate $\frac{\partial \mathbf{v}}{\partial t}$**

$$\frac{\partial v_1}{\partial t} = -\alpha(x_1^3 + x_1 x_2^2) \cdot (-\beta)\exp(-\beta t) = \alpha\beta(x_1^3 + x_1 x_2^2)\exp(-\beta t)$$

$$\frac{\partial v_2}{\partial t} = -\alpha(x_1^2 x_2 + x_2^3) \cdot (-\beta)\exp(-\beta t) = \alpha\beta(x_1^2 x_2 + x_2^3)\exp(-\beta t)$$

$$\frac{\partial v_3}{\partial t} = 0$$

**Step 3: Calculate $(\mathbf{v} \cdot \nabla)\mathbf{v}$**

First, we need the velocity gradient:
$$\frac{\partial v_1}{\partial x_1} = -\alpha(3x_1^2 + x_2^2)\exp(-\beta t)$$
$$\frac{\partial v_1}{\partial x_2} = -\alpha(2x_1 x_2)\exp(-\beta t)$$
$$\frac{\partial v_1}{\partial x_3} = 0$$

$$\frac{\partial v_2}{\partial x_1} = -\alpha(2x_1 x_2)\exp(-\beta t)$$
$$\frac{\partial v_2}{\partial x_2} = -\alpha(x_1^2 + 3x_2^2)\exp(-\beta t)$$
$$\frac{\partial v_2}{\partial x_3} = 0$$

All derivatives of $v_3$ are zero.

**Step 4: Calculate convective acceleration**

$$((\mathbf{v} \cdot \nabla)\mathbf{v})_1 = v_1\frac{\partial v_1}{\partial x} + v_2\frac{\partial v_1}{\partial y} + v_3\frac{\partial v_1}{\partial z}$$

$$= (-\alpha(x_1^3 + x_1 x_2^2)\exp(-\beta t))(-\alpha(3x_1^2 + x_2^2)\exp(-\beta t)) + \text{similar terms}$$

**Step 5: Evaluate at $(1,0,0)$ and $t = 0$**

At this point: $x_1 = 1$, $x_2 = 0$, $x_3 = 0$, and $\exp(-\beta \cdot 0) = 1$

$$\frac{\partial v_1}{\partial t}\bigg|_{(1,0,0),t=0} = \alpha\beta(1^3 + 1 \cdot 0^2) \cdot 1 = \alpha\beta$$

$$\frac{\partial v_2}{\partial t}\bigg|_{(1,0,0),t=0} = \alpha\beta(1^2 \cdot 0 + 0^3) \cdot 1 = 0$$

For convective terms:
$$v_1|_{(1,0,0),t=0} = -\alpha(1^3 + 1 \cdot 0^2) = -\alpha$$
$$v_2|_{(1,0,0),t=0} = -\alpha(1^2 \cdot 0 + 0^3) = 0$$

$$\frac{\partial v_1}{\partial x_1}\bigg|_{(1,0,0),t=0} = -\alpha(3 \cdot 1^2 + 0^2) = -3\alpha$$

Therefore:
$$((\mathbf{v} \cdot \nabla)\mathbf{v})_1 = (-\alpha)(-3\alpha) + 0 + 0 = 3\alpha^2$$
$$((\mathbf{v} \cdot \nabla)\mathbf{v})_2 = 0$$
$$((\mathbf{v} \cdot \nabla)\mathbf{v})_3 = 0$$

**Final result:**
$$a_1 = \alpha\beta + 3\alpha^2$$
$$a_2 = 0$$
$$a_3 = 0$$

## Problem 5 (Grade A-C)

A motion of a continuum body given by the equations

$$v_1 = \frac{3x_1}{1 + t}$$
$$v_2 = \frac{x_2}{1 + t}$$
$$v_3 = \frac{5x_3^2}{1 + t}$$

Assume that the reference configuration of the continuum body is at $t = 0$, with the consistency condition $X_i = x_i$.

a) Derive the particle path, i.e. the motion $x_i$

b) Compute the velocity components in terms of the material coordinates and time and the associated accelerations in the material and spatial description

### Solution to Problem 5

**Part (a): Derive the particle path (motion equations)**

To find the motion $x_i(X_i, t)$, we need to solve the system of ODEs:
$$\frac{dx_i}{dt} = v_i(x_i, t)$$

**For $x_1$:**
$$\frac{dx_1}{dt} = \frac{3x_1}{1 + t}$$

This is separable:
$$\frac{dx_1}{x_1} = \frac{3dt}{1 + t}$$

Integrating both sides:
$$\ln x_1 = 3\ln(1 + t) + C_1$$
$$x_1 = C_1(1 + t)^3$$

Using initial condition $x_1(X_1, 0) = X_1$:
$$X_1 = C_1(1)^3 \Rightarrow C_1 = X_1$$

Therefore: $x_1 = X_1(1 + t)^3$

**For $x_2$:**
$$\frac{dx_2}{dt} = \frac{x_2}{1 + t}$$

Similarly:
$$\frac{dx_2}{x_2} = \frac{dt}{1 + t}$$
$$\ln x_2 = \ln(1 + t) + C_2$$
$$x_2 = C_2(1 + t)$$

Using $x_2(X_2, 0) = X_2$: $C_2 = X_2$

Therefore: $x_2 = X_2(1 + t)$

**For $x_3$:**
$$\frac{dx_3}{dt} = \frac{5x_3^2}{1 + t}$$

This is separable:
$$\frac{dx_3}{x_3^2} = \frac{5dt}{1 + t}$$

Integrating:
$$-\frac{1}{x_3} = 5\ln(1 + t) + C_3$$
$$x_3 = \frac{-1}{5\ln(1 + t) + C_3}$$

Using $x_3(X_3, 0) = X_3$:
$$X_3 = \frac{-1}{5\ln(1) + C_3} = \frac{-1}{C_3}$$
$$C_3 = -\frac{1}{X_3}$$

Therefore: $x_3 = \frac{X_3}{1 - 5X_3\ln(1 + t)}$

**Motion equations:**
$$x_1 = X_1(1 + t)^3$$
$$x_2 = X_2(1 + t)$$
$$x_3 = \frac{X_3}{1 - 5X_3\ln(1 + t)}$$

**Part (b): Velocity and acceleration components**

**Material description of velocity:**
$$v_1 = \frac{dx_1}{dt} = 3X_1(1 + t)^2$$
$$v_2 = \frac{dx_2}{dt} = X_2$$
$$v_3 = \frac{dx_3}{dt} = \frac{5X_3^2}{(1 + t)(1 - 5X_3\ln(1 + t))^2}$$

**Material description of acceleration:**
$$a_1 = \frac{dv_1}{dt} = 6X_1(1 + t)$$
$$a_2 = \frac{dv_2}{dt} = 0$$
$$a_3 = \frac{dv_3}{dt} = \frac{5X_3^2}{(1 + t)^2(1 - 5X_3\ln(1 + t))^2} - \frac{50X_3^3}{(1 + t)^2(1 - 5X_3\ln(1 + t))^3}$$

**Spatial description of acceleration:**

We can also calculate acceleration using the convective derivative:
$$a_i = \frac{\partial v_i}{\partial t} + v_j\frac{\partial v_i}{\partial x_j}$$

Since the given velocities are already in spatial form, we need:
$$\frac{\partial v_1}{\partial t} = \frac{-3x_1}{(1 + t)^2}$$
$$\frac{\partial v_2}{\partial t} = \frac{-x_2}{(1 + t)^2}$$
$$\frac{\partial v_3}{\partial t} = \frac{-5x_3^2}{(1 + t)^2}$$

And the convective terms:
$$v_j\frac{\partial v_i}{\partial x_j} = v_1\frac{\partial v_i}{\partial x_1} + v_2\frac{\partial v_i}{\partial x_2} + v_3\frac{\partial v_i}{\partial x_3}$$

For $i = 1$: $\frac{\partial v_1}{\partial x_1} = \frac{3}{1+t}$, others are zero
For $i = 2$: $\frac{\partial v_2}{\partial x_2} = \frac{1}{1+t}$, others are zero  
For $i = 3$: $\frac{\partial v_3}{\partial x_3} = \frac{10x_3}{1+t}$, others are zero

**Final spatial acceleration:**
$$a_1 = \frac{-3x_1}{(1 + t)^2} + \frac{3x_1}{1+t} \cdot \frac{3}{1+t} = \frac{6x_1}{(1 + t)^2}$$
$$a_2 = \frac{-x_2}{(1 + t)^2} + \frac{x_2}{1+t} \cdot \frac{1}{1+t} = 0$$
$$a_3 = \frac{-5x_3^2}{(1 + t)^2} + \frac{5x_3^2}{1+t} \cdot \frac{10x_3}{1+t} = \frac{5x_3^2(10x_3 - 1)}{(1 + t)^2}$$

In [23]:
# Problem 5 - Complete solution verification
import sympy as sp

# Define symbols
X1, X2, X3, x1, x2, x3, t = sp.symbols('X1 X2 X3 x1 x2 x3 t', real=True, positive=True)

print("PROBLEM 5 SOLUTION")
print("="*50)

# Given velocity field (spatial description)
v1_spatial = 3*x1/(1 + t)
v2_spatial = x2/(1 + t)
v3_spatial = 5*x3**2/(1 + t)

print("Given spatial velocity field:")
print(f"v1 = {v1_spatial}")
print(f"v2 = {v2_spatial}")
print(f"v3 = {v3_spatial}")

print("\nPART (A): DERIVING MOTION EQUATIONS")
print("-"*40)

# Solve ODEs to find motion equations
print("Solving dx1/dt = 3x1/(1+t):")
print("∫dx1/x1 = ∫3dt/(1+t)")
print("ln(x1) = 3ln(1+t) + C1")
print("x1 = C1(1+t)³")
print("Using x1(0) = X1: C1 = X1")

x1_motion = X1*(1 + t)**3
print(f"Motion equation: x1 = {x1_motion}")

print("\nSolving dx2/dt = x2/(1+t):")
x2_motion = X2*(1 + t)
print(f"Motion equation: x2 = {x2_motion}")

print("\nSolving dx3/dt = 5x3²/(1+t):")
print("∫dx3/x3² = ∫5dt/(1+t)")
print("-1/x3 = 5ln(1+t) + C3")
print("Using x3(0) = X3: C3 = -1/X3")
x3_motion = X3/(1 - 5*X3*sp.log(1 + t))
print(f"Motion equation: x3 = {x3_motion}")

print("\nPART (B): VELOCITY AND ACCELERATION")
print("-"*40)

# Material description of velocity
v1_material = sp.diff(x1_motion, t)
v2_material = sp.diff(x2_motion, t)
v3_material = sp.diff(x3_motion, t)

print("Velocity - Material description:")
print(f"v1 = {v1_material}")
print(f"v2 = {v2_material}")
print(f"v3 = {sp.simplify(v3_material)}")

# Material description of acceleration
a1_material = sp.diff(v1_material, t)
a2_material = sp.diff(v2_material, t)
a3_material = sp.diff(v3_material, t)

print("\nAcceleration - Material description:")
print(f"a1 = {a1_material}")
print(f"a2 = {a2_material}")
print(f"a3 = {sp.simplify(a3_material)}")

# Spatial description of acceleration using convective derivative
print("\nAcceleration - Spatial description:")
print("Using a = ∂v/∂t + (v·∇)v")

# Time derivatives
dv1_dt_spatial = sp.diff(v1_spatial, t)
dv2_dt_spatial = sp.diff(v2_spatial, t)
dv3_dt_spatial = sp.diff(v3_spatial, t)

print(f"∂v1/∂t = {dv1_dt_spatial}")
print(f"∂v2/∂t = {dv2_dt_spatial}")
print(f"∂v3/∂t = {dv3_dt_spatial}")

# Spatial derivatives
dv1_dx1 = sp.diff(v1_spatial, x1)
dv2_dx2 = sp.diff(v2_spatial, x2)
dv3_dx3 = sp.diff(v3_spatial, x3)

print(f"\n∂v1/∂x1 = {dv1_dx1}")
print(f"∂v2/∂x2 = {dv2_dx2}")
print(f"∂v3/∂x3 = {dv3_dx3}")

# Convective acceleration (only diagonal terms are non-zero)
conv_a1 = v1_spatial * dv1_dx1
conv_a2 = v2_spatial * dv2_dx2
conv_a3 = v3_spatial * dv3_dx3

print(f"\nConvective acceleration:")
print(f"v1∂v1/∂x1 = {conv_a1}")
print(f"v2∂v2/∂x2 = {conv_a2}")
print(f"v3∂v3/∂x3 = {conv_a3}")

# Total spatial acceleration
a1_spatial_total = dv1_dt_spatial + conv_a1
a2_spatial_total = dv2_dt_spatial + conv_a2
a3_spatial_total = dv3_dt_spatial + conv_a3

print(f"\nTotal acceleration - Spatial description:")
print(f"a1 = {sp.simplify(a1_spatial_total)}")
print(f"a2 = {sp.simplify(a2_spatial_total)}")
print(f"a3 = {sp.simplify(a3_spatial_total)}")

# Verify consistency by substituting motion equations into spatial acceleration
print(f"\nVERIFICATION:")
print("Substituting motion equations into spatial acceleration:")
a1_check = a1_spatial_total.subs(x1, x1_motion)
a2_check = a2_spatial_total.subs(x2, x2_motion)

print(f"a1 (material) = {a1_material}")
print(f"a1 (spatial→material) = {sp.simplify(a1_check)}")
print(f"Match: {sp.simplify(a1_material - a1_check) == 0}")

print(f"\na2 (material)       = {a2_material}")
print(f"a2 (spatial→material) = {sp.simplify(a2_check)}")
print(f"Match: {sp.simplify(a2_material - a2_check) == 0}")

PROBLEM 5 SOLUTION
Given spatial velocity field:
v1 = 3*x1/(t + 1)
v2 = x2/(t + 1)
v3 = 5*x3**2/(t + 1)

PART (A): DERIVING MOTION EQUATIONS
----------------------------------------
Solving dx1/dt = 3x1/(1+t):
∫dx1/x1 = ∫3dt/(1+t)
ln(x1) = 3ln(1+t) + C1
x1 = C1(1+t)³
Using x1(0) = X1: C1 = X1
Motion equation: x1 = X1*(t + 1)**3

Solving dx2/dt = x2/(1+t):
Motion equation: x2 = X2*(t + 1)

Solving dx3/dt = 5x3²/(1+t):
∫dx3/x3² = ∫5dt/(1+t)
-1/x3 = 5ln(1+t) + C3
Using x3(0) = X3: C3 = -1/X3
Motion equation: x3 = X3/(-5*X3*log(t + 1) + 1)

PART (B): VELOCITY AND ACCELERATION
----------------------------------------
Velocity - Material description:
v1 = 3*X1*(t + 1)**2
v2 = X2
v3 = 5*X3**2/((t + 1)*(5*X3*log(t + 1) - 1)**2)

Acceleration - Material description:
a1 = 3*X1*(2*t + 2)
a2 = 0
a3 = X3**2*(-25*X3*log(t + 1) - 50*X3 + 5)/((t + 1)**2*(5*X3*log(t + 1) - 1)**3)

Acceleration - Spatial description:
Using a = ∂v/∂t + (v·∇)v
∂v1/∂t = -3*x1/(t + 1)**2
∂v2/∂t = -x2/(t + 1)**2
∂v3/∂t = -5*